yaptide/routes/common_sim_routes.py

```diff
@@ -168,12 +172,21 @@ def post():
         if decoded_token != sim_id:
             return yaptide_response(message="Invalid update key", code=400)
 
-        for estimator_dict in payload_dict["estimators"]:
+        for i, estimator_dict in enumerate(payload_dict["estimators"]):
             # We forsee the possibility of the estimator being created earlier as element of partial results
             estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_dict["name"])
-
             if not estimator:
-                estimator = EstimatorModel(name=estimator_dict["name"], simulation_id=simulation.id)
+                file_name = estimator_dict["name"]
+                if simulation.sim_type == SimulationType.FLUKA.value and len(simulation.inputs) > 0:
```

Comment:
I would like to avoid such if-ology at any cost. It introduces unnecessary complexity.

Reply:
Also checking for presence of `simulation.input`  is much less readable than check for simulation type in metadata.

From that code is not obvious that we are considering cases where simulation is defined by input files and by 3D threejs editor.

Having such an `if` here would lead to much cleaner code.

In case Editor projects, for both Fluka and SHIELD-HIT12A entries in the input files are generated in the same order as they appear in the Editor"
https://github.com/yaptide/converter/blob/fa381ece463b8cc0ff6ac2b199c76f35a60bcae6/converter/fluka/cards/scoring_card.py#L171-L178
https://github.com/yaptide/converter/blob/fa381ece463b8cc0ff6ac2b199c76f35a60bcae6/converter/shieldhit/parser.py#L277-L289
The difference is when the output files are generated and the JSON files with results are prepared. Then again in both cases the files are sorted alphabetically and included in such order in the results JSON.
As Fluka is using sequential numbers (21,22,23...) for the output that order is the same as order of Output cards in the editor. For SHIELD-HIT12A that order is not the same, as User may define first the `b` output then `a` output, so they appear in `BA` sequence in input and in `AB` sequence in results.


So we could have the following logic:

if "input files":
  estimator_name = filename
elif "editor"
  some logic here which distinguish Fluka from SHIELD-HIT12A
else
 handle error

Reply:
> Then again in both cases the files are sorted alphabetically and included in such order in the results JSON.

Are you sure about that? 

![order](https://github.com/user-attachments/assets/6192e7af-d5d5-4669-ba55-c288fada3c11)
It looks like the estimators are appended in random order.

https://github.com/DataMedSci/pymchelper/blob/e862c6ef638fa9470286c7154fa2ef9b782cc1c2/pymchelper/input_output.py#L192
I think this is because the glob function returns JSON files in random order. If we modify the code to use sorted, like this:
`list_of_matching_files = sorted(glob(pattern))`

It would be possible to match names same way for fluka and shieldhit

```
        if simulation.input_type == InputType.EDITOR.value:
            outputs = simulation.inputs[0].data["input_json"]["scoringManager"]["outputs"]
            estimator_names = sorted([output["name"] for output in outputs])

        for i, estimator_dict in enumerate(payload_dict["estimators"]):
            # We forsee the possibility of the estimator being created earlier as element of partial results
            estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_dict["name"])
            if not estimator:
                file_name = estimator_dict["name"]
                if simulation.input_type == InputType.FILES.value:
                    estimator_name = file_name
                elif simulation.input_type == InputType.EDITOR.value:
                    estimator_name = estimator_names[i]
```

Reply:
> It would be possible to match names same way for fluka and shieldhit
> 
> ```
>         if simulation.input_type == InputType.EDITOR.value:
>             outputs = simulation.inputs[0].data["input_json"]["scoringManager"]["outputs"]
>             estimator_names = sorted([output["name"] for output in outputs])
> 
>         for i, estimator_dict in enumerate(payload_dict["estimators"]):
>             # We forsee the possibility of the estimator being created earlier as element of partial results
>             estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_dict["name"])
>             if not estimator:
>                 file_name = estimator_dict["name"]
>                 if simulation.input_type == InputType.FILES.value:
>                     estimator_name = file_name
>                 elif simulation.input_type == InputType.EDITOR.value:
>                     estimator_name = estimator_names[i]
> ```


Does it mean that with that approach we could get rid of reordering estimators in UI as described here:
https://github.com/yaptide/yaptide/pull/1243#issuecomment-2525537643 ?

Reply:
Another question - would it be possible to implement that reordering in alembic as well to maintain backward compatibility ?